### PR TITLE
CTCP-4747: Create new procedure page (feature branch)

### PR DESCRIPTION
**Pull request by justega247**

### CODE REVIEW

**Review by kevincoatsworth — APPROVED** [review]

**Review by prashanthmamidi — DISMISSED** [review]

**Review by kenneth-rayner — APPROVED** [review]

**Comment by kenneth-rayner** [comment]
Screenshot from 2024-07-10 15-12-31
Screenshot from 2024-07-10 15-12-10

File: app/views/NewAuthYesNoView.scala.html [hunk 1]
```diff
@@ -40,10 +40,21 @@
...
+                additionalHtml = Html(
+                        s"""
+                        <p>${messages("newAuthYesNo.paragraph").toParagraph}</p>
+                        <p class = "govuk-hint">${messages("newAuthYesNo.hint")}</p>
+                            <ul class="govuk-list govuk-list--bullet govuk-hint">
+                                  <li>${messages("newAuthYesNo.bullet1")}</li>
+                                  <li>${messages("newAuthYesNo.bullet2")}</li>
+                                  <li>${messages("newAuthYesNo.bullet3")}</li>
+                            </ul>
+                        """
```
Comment (jackheslop96) [inline app/views/NewAuthYesNoView.scala.html:55]: Why aren't we providing the hint text to the hint param?

File: test/views/NewAuthYesNoViewSpec.scala [hunk 2]
```diff
@@ -37,6 +37,10 @@ class NewAuthYesNoViewSpec extends YesNoViewBehaviours {
 
   behave like pageWithHeading()
 
+  behave like pageWithContent("p", "This allows you to make unloading remarks without physically unloading the goods.")
+
+  behave like pageWithHint("You can only use the revised unloading procedure if:")
+
```
Comment (jackheslop96) [inline test/views/NewAuthYesNoViewSpec.scala:43]: Could also use https://github.com/hmrc/manage-transit-movements-unloading-frontend/blob/c5f5169be743b5bc44514656a8a3a274b45355c0/test/views/behaviours/ViewBehaviours.scala#L239-L244

Something like 
```
behave like pageWithList(
  "govuk-list--bullet",
  "all items have the same office of destination",
  "there are no broken seals",
  "there are no discrepancies between the transit movement and unloading permission"
)
```

File: app/views/NewAuthYesNoView.scala.html (hunk 1 above)
Comment (jackheslop96) [reply app/views/NewAuthYesNoView.scala.html:55]: Something like

```
@(form: Form[Boolean],mrn: MovementReferenceNumber, arrivalId: ArrivalId, mode: Mode)(implicit request: Request[_], messages: Messages)

@html = {
    <p>@messages("newAuthYesNo.hint")</p>
    <ul class="govuk-list govuk-list--bullet govuk-hint">
        <li>@messages("newAuthYesNo.bullet1")</li>
        <li>@messages("newAuthYesNo.bullet2")</li>
        <li>@messages("newAuthYesNo.bullet3")</li>
    </ul>
}

@mainTemplate(
    title = messages("newAuthYesNo.title"),
    showBackLink = true
) {

        @errorSummary(form.errors)

        @formHelper(action = controllers.routes.NewAuthYesNoController.onSubmit(arrivalId, mode), Symbol("autoComplete") -> "off") {

        @inputYesNo(
            field = form("value"),
            yesNoType = YesNoWithAdditionalHtml(
                heading = messages("newAuthYesNo.heading"),
                caption = Some(messages("site.caption", mrn.toString)),
                additionalHtml = messages("newAuthYesNo.paragraph").toParagraph
            ),
            hint = Some(HtmlContent(html))
        )

        @button(attributes = Map("id" -> "submit"))
    }
}
```

File: test/views/NewAuthYesNoViewSpec.scala (hunk 2 above)
Comment (kenneth-rayner) [reply test/views/NewAuthYesNoViewSpec.scala:55]: @jackheslop96 done

File: app/views/NewAuthYesNoView.scala.html (hunk 1 above)
Comment (kenneth-rayner) [reply app/views/NewAuthYesNoView.scala.html]: @jackheslop96 done  please re-review

**Review by jackheslop96 — APPROVED** [review]

**Review by Reece-Carruthers — DISMISSED** [review]

**Review by kevincoatsworth — APPROVED** [review]

**Review by vajarova — APPROVED** [review]

**Pull request merged**